src/bun.js/bindings/bindings.zig

```diff
@@ -692,6 +692,11 @@ pub const ZigString = extern struct {
     }
 
     pub fn toExternalU16(ptr: [*]const u16, len: usize, global: *JSGlobalObject) JSValue {
+        if (len > String.max_length) {
+            bun.default_allocator.free(ptr[0..len]);
```

Comment:
I think the caller should free this. It's not guaranteed to be allocated with `bun.default_allocator`

Reply:
https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/src/bun.js/webcore/blob.zig#L4915-L4919

Reply:
it is guaranteed. note how it never passes an allocator